SPEC.md

```diff
@@ -98,61 +98,113 @@ A build tool is something which reads one or more input artifacts and writes one
 ```
 
-### GitBOM identifier
+The child artifact redords must be written to the GitBOM Document in lexical order.
 
-For a given artifact, the GitBOM ID of that artifact is simply the git object id of it's GitBOM Document.
+The artifact id and GitBOM document id must both be of the identifier type declared in the GitBOM document header.
 
 ### GitBOM Identifier Embedding
 
-Each build tool should embed the GitBOM ID of its output artifact(s) into the output artifact in a manner:
+Each build tool should embed into the output artifact a new line delimited, lexically ordered, list of GitBOM identifiers for each mandatory identifier type in a manner:
 
-1.  Appropriate to the type of artifact
-2.  Generally agreed upon for that artifact
+1. Appropriate to the type of artifact
+2. Generally agreed upon for that artifact
 
 ### GitBOM Document Construction by a Build Tool
 
-When a build tool reads an artifact, it should for each input file:
-1. Compute the git object id of the input file - ${artifact identifier}
-2. Examine the artifact for an embedded GitBOM identifier - ${gitbom idenfitier}
-3. If the artifact contains a GitBOM identifer it construct a record:
-   ```
-   blob⎵${artifact identifier}⎵bom⎵${GitBOM identifier}\n
-   ```
-4. Otherwise construct a record:
-   ```
-   blob⎵${artifact identifier}\n
-   ```
-
-Concatenate all records together and apply lexical ordering by line to create the GitBOM document of the
-output artifact.
+A build tool creating an output artifact must compute a GitBOM document of each mandatory artifact id type.
+
+For each input artifact the build tool must:
+
+1. Compute the git object id of the input - ${artifact identifier}
+2. Examine the input for an embedded GitBOM identifier - ${gitbom idenfitier}
+
+The build tool must persist a GitBOM document using the ${artifact identifier} and ${gitbom identifier} for each input.
 
 ### GitBOM Document persistence by a Build Tool
 
-When persisting an output artifact to a file system, the build tool should create a subdirectory of the directory to which the output artifact is being written of the form:
-```.bom/objects/``` and write out the GitBOM document to ```.bom/objects/${gitoid:0:2}/${gitoid:2:}``` where ```${gitoid}```
-is the git object id of the GitBOM document.
+Each build tool should persist all GitBOM documents it generates.
+
+Generically, GitBOM persistence can be thought of as having an abstract API:
+```
+WriteGitBOM(GitBOM id,GitBOM contents) returns error
+ReadGitBOM(GitBOM id) returns GitBOM Contents
+```
+
+WriteGitBOM should check that GitBOM id matches GitBOM contents and return an error if it does not.
+
+####  GitBOM Document persistence by a Build Tool to its local filesystem
+If a build tool persists GitBOM information to its local filesystem, the build tool should write out the GitBOM document to ```${GITBOM_DIR}/objects/${GitBOMID:0:2}/${GitBOMID:2:}``` where ```${artifact id}``` is the GitBOM id in lowercase hexidecimal.
```

Comment:
Should it be ${GITBOM_DIR}/.gitbom/objects as the current prototypes have it ?

Reply:
@bharsesh I'm open to that discussion... but I'm curious why not just set ${GITBOM_DIR} to include the .gitbom/ portion if that is what the user desires?

Reply:
If .gitbom is user specific, then its ok. It could well be .bom or .gitbom whatever the user desires. I pointed this out mainly for consistency as we switched between .bom and .gitbom in the tools.